src/Controls/src/Core/Platform/Android/Extensions/FormattedStringExtensions.cs

```diff
@@ -26,7 +26,28 @@ public static SpannableString ToSpannableString(this Label label)
 				label.TextTransform,
 				label.TextDecorations);
 
+		// TODO: this overload must be removed in net7.0 and replaced with the one below
 		public static SpannableString ToSpannableString(
+			this FormattedString formattedString,
+			IFontManager fontManager,
+			TextPaint? textPaint = null,
```

Comment:
Using TextPaint was incorrect as it did not represent the span size but the label. So the label would have a tiny font size and the span huge, but then layouts would measure line heights using the tine label size.